src/Persistent/Pure.hs

```diff
@@ -463,14 +465,15 @@ moveIdeasToLocation ideaIds location =
 setTopicPhase :: AUID Topic -> Phase -> AUpdate ()
 setTopicPhase tid phase = modifyTopic tid $ topicPhase .~ phase
 
-addTopic :: AddDb Topic
-addTopic pt = do
+addTopic :: Timestamp -> AddDb Topic
+addTopic now pt = do
     t <- addDb dbTopicMap pt
     dbFrozen <- liftAQuery $ view dbFreeze
-    -- FIXME: the invariant that topics start in Ref phases is now spread
-    -- across a couple of different places. To merge it, or when it no longer
-    -- holds, use the full @phaseChange@ machinery (requiring @now@, etc.).
-    when (dbFrozen == Frozen) $ setTopicPhase (t ^. _Id) (PhaseRefFrozen 0)
+    when (dbFrozen == Frozen) $ do
+        case LifeCycle.phaseTrans (t ^. topicPhase) (LifeCycle.PhaseFreeze now) of
+            Just (phase'@(PhaseRefFrozen _), [])
+                -> setTopicPhase (t ^. _Id) phase'
+            bad -> assert False . error $ "addTopic: internal error: " <> show bad
```

Comment:
Yeah, thank you for that. If only I wrote `PhaseRefFrozen  {_refPhaseLeftover = 0}` instead of `PhaseRefFrozen 0`, I'd spot my bug outright. :) This could be fixed by fetching the current Duration and inserting it in place of 0, but that's not KISS any more, so the proper, general and extensible way you implemented beats it.